code/modules/paperwork/paper.dm

```diff
@@ -351,25 +351,28 @@
 		return TRUE
 	return ..()
 
-/obj/item/proc/burn_paper_product_attackby_check(obj/item/I, mob/living/user, bypass_clumsy)
-	var/ignition_message = I.ignition_effect(src, user)
+/obj/item/proc/burn_paper_product_attackby_check(obj/item/attacking_item, mob/living/user, bypass_clumsy = FALSE)
+	//can't be put on fire!
+	if((resistance_flags & FIRE_PROOF) || !(resistance_flags & FLAMMABLE))
+		return
+	var/ignition_message = attacking_item.ignition_effect(src, user)
 	if(!ignition_message)
 		return
 	. = TRUE
```

Comment:
this makes the whole proc really fucking silly so while we're here let's fix this in the best way possible

```suggestion
```